src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -266,47 +267,25 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
             return builder.ToImmutableAndFree();
         }
 
-        private BoundPattern BindLengthPatternClause(
-            LengthPatternClauseSyntax node, TypeSymbol inputType,
-            bool permitDesignations, out PropertySymbol? lengthProperty, ref bool hasErrors, BindingDiagnosticBag diagnostics)
-        {
-            if (!TryLookupLengthOrCount(node, inputType, out lengthProperty) && !inputType.IsErrorType())
-            {
-                Error(diagnostics, ErrorCode.ERR_UnsupportedTypeForLengthPattern, node, inputType.ToDisplayString());
-                hasErrors = true;
-            }
-            return BindPattern(node.Pattern, Compilation.GetSpecialType(SpecialType.System_Int32), ExternalScope, permitDesignations, hasErrors, diagnostics);
-        }
-
-        private bool TryLookupLengthOrCount(SyntaxNode node, TypeSymbol inputType, out PropertySymbol? lengthProperty)
-        {
-            Debug.Assert(node.IsKind(SyntaxKind.ListPatternClause) || node.IsKind(SyntaxKind.LengthPatternClause));
-            var lookupResult = LookupResult.GetInstance();
-            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
-            bool found = TryLookupLengthOrCount(inputType, lookupResult, out lengthProperty, ref useSiteInfo);
-            lookupResult.Free();
-            return found;
-        }
-
-        private BoundListPatternClause BindListPatternClause(
-            PropertyPatternClauseSyntax node,
+        private BoundListPattern BindListPattern(
+            ListPatternSyntax node,
             TypeSymbol inputType,
+            uint inputValEscape,
             bool permitDesignations,
-            ref PropertySymbol? lengthProperty,
-            ref bool hasErrors,
+            bool hasErrors,
             BindingDiagnosticBag diagnostics)
         {
-            Debug.Assert(node.IsKind(SyntaxKind.ListPatternClause));
-
             TypeSymbol elementType;
             BoundIndexerAccess? indexerAccess = null;
             PropertySymbol? indexerSymbol = null;
-            bool isUnsupported = false;
+            PropertySymbol? lengthProperty;
             if (inputType.IsSZArray())
             {
                 elementType = ((ArrayTypeSymbol)inputType).ElementType;
+                lengthProperty = (PropertySymbol?)((MethodSymbol?)GetWellKnownTypeMember(WellKnownMember.System_Array__get_Length, diagnostics, syntax: node))?.AssociatedSymbol;
```

Comment:
Let's include a test with missing `System_Array__get_Length`

Reply:
We already have ListPattern_MissingMembers_ArrayLength